src/main/java/seedu/address/model/house/Condominium.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.model.house;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+
+/**
+ * Represents a Condominium
+ */
+public class Condominium implements House {
+
+    public final PostalCode postalCode;
+    public final Street street;
+    public final UnitNumber unitNumber;
+    public final Block block;
+    public final Level level;
+
+    /**
+     * Constructs a Condominium.
+     *
+     * @param unitNumber The unit number of the house.
+     * @param postalCode The postal code of the house.
+     * @param street     The street of the house.
+     * @param block      The block of the house.
+     * @param level      The level of the house.
+     */
+    public Condominium(Level level, PostalCode postalCode, Street street, UnitNumber unitNumber, Block block) {
+        this.postalCode = postalCode;
+        this.street = street;
+        this.unitNumber = unitNumber;
+        this.block = block;
+        this.level = level;
+    }
+
+    /**
+     * Constructs a Condominium with no block
+     *
+     * @param unitNumber The unit number of the house.
+     * @param postalCode The postal code of the house.
+     * @param street     The street of the house.
+     * @param level      The level of the house.
+     */
+    public Condominium(Level level, PostalCode postalCode, Street street, UnitNumber unitNumber) {
+        this.postalCode = postalCode;
+        this.street = street;
+        this.unitNumber = unitNumber;
+        this.block = new Block("N/A");
+        this.level = level;
+    }
```

Comment:
I think you can remove this overloaded constructor.

Since block is optional, I think you can refer to how Tag is implemented as they're both optional on creation.